tests/test_ssh2.c

```diff
@@ -24,6 +24,11 @@
 #include <stdio.h>
 #include <stdlib.h>  /* for getenv() */
 
+#if defined(HAVE_PTHREAD_H) && defined(HAVE_SIGNAL_H)
+#include <pthread.h>
+#include <signal.h>
+#endif
```

Comment:
```suggestion
#if defined(HAVE_PTHREAD_H)
#include <pthread.h>
#include <signal.h>

static pthread_t main_thread;
#endif
```
`signal.h` is standard C89, let's drop the guard. Also from the rest of the code.

`pthread.h` needs detection in autotools too.

These said, I'm also fine if we don't add this sample/test code to the test.